### ISSUE TITLE
Conformance test for SYCL is not working.

### ISSUE THREAD

**Issue by ankitbmishra** [KhronosGroup/SYCL-CTS#925]
I tried to build conformance test using the code repository given here, for SYCL compilers:
1. DPC++
2. Our own SYCL compiler

But, the build failed for both the above SYCL compilers.
For DPC++  it is giving error multiple time (screenshot attached):
```bash
/storage/ankit/SYCL-CTS/build/tests/vector_api/vector_api_bool.cpp:86:57: error: no type named 'value_type' in 'sycl::vec<bool, 1>'
   86 |       CHECK(std::is_same_v<typename sycl::vec<bool, 1>::value_type, bool>);
```
Please provide solution for this issue.

Reference screenshots of DPC++ compiler conformance testing:
```bash
cmake -S . -B build -DSYCL_IMPLEMENTATION=DPCPP
```
![Screenshot from 2024-08-16 07-59-16](https://github.com/user-attachments/assets/e70feaeb-4403-40b4-b01d-b81e292b3819)

```bash
cmake --build ./build -j24
```
![Screenshot from 2024-08-16 07-59-05](https://github.com/user-attachments/assets/50d8f34b-33ff-40e7-a3d7-e0a0cea5e43e)

ERROR:
![Screenshot from 2024-08-16 07-57-18](https://github.com/user-attachments/assets/8fc11c10-034c-42a6-800e-7504fc0cab89)

**Comment by bader** [comment]
What DPC++ version do you use? CTS CI uses 20a088e1231c4ac85fd74c0de79c563977d2f38c, which successfully compiles all tests.

**Comment by ankitbmishra** [comment]
Can you please share link of particular dpc++ version that you mentioned in reply.

**Comment by bader** [comment]
Here is the link: https://github.com/intel/llvm/commit/20a088e1231c4ac85fd74c0de79c563977d2f38c
If you are looking for pre-built binaries of the compiler you can find them here - https://github.com/intel/llvm/releases/tag/nightly-2024-07-11.

**Comment by ankitbmishra** [comment]
1. exported above mentioned binaries.(https://github.com/intel/llvm/releases/tag/nightly-2024-07-11)
2. tried to get conformance test but still unable to build.(reference in image)
![Screenshot from 2024-08-27 15-42-47](https://github.com/user-attachments/assets/6db68286-c5c1-4c0c-b1d8-61818824f292)

Please help. 
thank you.

**Comment by psalz** [comment]
>     2. tried to get conformance test but still unable to build.(reference in image)

The error message says what you need to do!

**Comment by ankitbmishra** [comment]
> > ```
> > 2. tried to get conformance test but still unable to build.(reference in image)
> > ```
> 
> The error message says what you need to do!

As per the documentation, it should work without any modification! (https://github.com/KhronosGroup/SYCL-CTS)

**Comment by psalz** [comment]
> > > ```
> > > 2. tried to get conformance test but still unable to build.(reference in image)
> > > ```
> > 
> > 
> > The error message says what you need to do!
> 
> As per the documentation, it should work without any modification! (https://github.com/KhronosGroup/SYCL-CTS)

As per the documentation, you should do a recursive checkout:

> Configuration and compilation then follow standard CMake procedures. Begin by cloning this repository and its submodules:
> `git clone --recurse-submodules https://github.com/KhronosGroup/SYCL-CTS.git`

**Comment by ankitbmishra** [comment]
> > > > ```
> > > > 2. tried to get conformance test but still unable to build.(reference in image)
> > > > ```
> > > 
> > > 
> > > The error message says what you need to do!
> > 
> > 
> > As per the documentation, it should work without any modification! (https://github.com/KhronosGroup/SYCL-CTS)
> 
> As per the documentation, you should do a recursive checkout:
> 
> > Configuration and compilation then follow standard CMake procedures. Begin by cloning this repository and its submodules:
> > `git clone --recurse-submodules https://github.com/KhronosGroup/SYCL-CTS.git`

As per the documentation,
(1) checked cmake version
(2) checked python version
(3) checked sycl implementation

Still it was giving error because SYCL-CTS/vendor/Catch2 to be added manually (link: https://github.com/catchorg/Catch2/tree/d548be26e3c54abc7e0b076319057d334e3ac589)

That is done and cmake command worked successfully.
```bash
cmake -S . -B build -DSYCL_IMPLEMENTATION=DPCPP
```
after this, as per the documentation next command given:
```bash
cmake --build ./build
```
![Screenshot from 2024-08-27 16-40-43](https://github.com/user-attachments/assets/dabd44c2-5d83-4b7f-b02a-455a63491d88)

But still the build is not complete and it is showing multiple errors!(reference in images)
![Screenshot from 2024-08-27 16-38-59](https://github.com/user-attachments/assets/3b873455-430a-43df-9de5-64259b239869)
![Screenshot from 2024-08-27 16-40-43](https://github.com/user-attachments/assets/34bcd52a-1b38-4144-8d51-b91147b04a77)

**Comment by ankitbmishra** [comment]
> > > > ```
> > > > 2. tried to get conformance test but still unable to build.(reference in image)
> > > > ```
> > > 
> > > 
> > > The error message says what you need to do!
> > 
> > 
> > As per the documentation, it should work without any modification! (https://github.com/KhronosGroup/SYCL-CTS)
> 
> As per the documentation, you should do a recursive checkout:
> 
> > Configuration and compilation then follow standard CMake procedures. Begin by cloning this repository and its submodules:
> > `git clone --recurse-submodules https://github.com/KhronosGroup/SYCL-CTS.git`

As per the documentation,
(1) checked cmake version
(2) checked python version
(3) checked sycl implementation

Still it was giving error because SYCL-CTS/vendor/Catch2 to be added manually (link: https://github.com/catchorg/Catch2/tree/d548be26e3c54abc7e0b076319057d334e3ac589)

That is done and cmake command worked successfully.
```bash
cmake -S . -B build -DSYCL_IMPLEMENTATION=DPCPP
```
after this, as per the documentation next command given:
```bash
cmake --build ./build
```
![Screenshot from 2024-08-27 16-40-43](https://github.com/user-attachments/assets/dabd44c2-5d83-4b7f-b02a-455a63491d88)

But still the build is not complete and it is showing multiple errors!(reference in images)
![Screenshot from 2024-08-27 16-38-59](https://github.com/user-attachments/assets/81a4cfe0-2928-49d5-9e30-554869ac094a)
![Screenshot from 2024-08-27 16-39-15](https://github.com/user-attachments/assets/12737c00-e3d4-4446-96fb-1fa63ee1fcec)

**Comment by keryell** [comment]
I guess we can close this as there have been several versions of DPC++ passing the CTS since.

**Comment by bader** [comment]
I agree with @keryell. It looks like environment issue. According to the error messages in the attached screenshots, incompatible DPC++ compiler has been used to compile SYCL-CTS sources.
